### DISCUSSION TITLE
Remote log output / log system?

### DISCUSSION THREAD

**Discussion by terminar** [sboesen/remotely-sync#78] [Ideas]
Due to the current problems I have with remotely-save I'm thinking about implementing some remote log receiver.
Idea is just simple, 

1. create some "stupid" receiver on a webserver or application which receives sort of syslog type (or plain text like) console messages with some timestamp.

2. implement the sender in remotely-sync

3. (maybe only in my local installation) put useful amount of log step messages into the code to see where it crashes

Do you have something planned with specific software components? Or is this something you wouldn't merge?

**Comment by sboesen** [comment]
I like the idea, not too complex, so for debugging it could make sense - to clarify, would a plugin like https://github.com/czottmann/obsidian-logstravaganza solve your issues?

I assume you are running into issues on mobile where you can't pull up the application logs as easily (at least on iOS).

**Reply by terminar** [reply]
Problem is that even the log and database dump files from remotely-save lead to a crash because of a too big size. Obsidian starts, tries to index the files and then crashes. I had to remove the folder with the logs, now I was able to start again without crashes (only on sync).

so my idea was:
- not to write anything on the device for the logs
- Stream/send logs / maybe wrapped console.log to a debug receiver

Btw. Is there an official/easy way to update the plugin files after rebuild to mobile other than using BART or some manual copy?

**Reply by sboesen** [reply]
> Problem is that even the log and database dump files from remotely-save lead to a crash because of a too big size.

Do you need these to be synced? You can set max file size for syncing in Remotely Sync's settings:
<img width="544" alt="Screenshot 2023-12-22 at 9 26 10 AM" src="https://github.com/sboesen/remotely-sync/assets/248703/754567f1-163f-437e-b195-3ce9b22fe971">

> Btw. Is there an official/easy way to update the plugin files after rebuild to mobile other than using BART or some manual copy?

Not that I'm aware of, it's pretty painful. Options for troubleshooting crashes on launch are also pretty limited on mobile, from my own exploration and reading Obsidian forum posts.

**Comment by fyears** [comment]
(author of remotely-save here and i am defending myself)

logging was treated very seriously in the very beginning of development. logs are saved and can exported to .md files by clicking a button in settings. after all i myself also need to read the logs from ios!

remote logger is cool and i never thought about that. if the privacy issue is addressed i am happy to merge the feature into remotely-save.

and by debugging on ios, there is a way to copy the main.js manually directly from computer without bart but yes it's painful.

**Reply by sboesen** [reply]
> (author of remotely-save here and i am defending myself)

> logging was treated very seriously in the very beginning of development. logs are saved and can exported to .md files by clicking a button in settings. after all i myself also need to read the logs from ios!

Thanks for the reply - I definitely agree with this. I think the biggest gap with logging on iOS I have seen is when the plugin crashes on load (when I updated some of the npm dependencies I was getting errors enabling the plugin only on mobile, which was challenging to troubleshoot). Thankfully Android is easier to work with, and I believe this helped me isolate the issue. To be clear this definitely is not an issue with Remotely {Save,Sync} and is something that the Obsidian team would need to add better support for.

It sounds like @terminar was running into crashing due to the size of logs on mobile which I haven't experienced myself. Remote logging may help for those situations if they continue occurring.
